### PR TITLE
Changing the data model for filing service

### PR DESCRIPTION
**Pull request by sampankumar**
__Short description outlining key changes/additions__

Changing the data model for filing service

__JIRA Ticket Number__

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java [hunk 1]
```diff
@@ -11,7 +11,7 @@ public final class ACSP {
     private String paymentMethod;
 
     @JsonProperty("correspondence_address")
```
Comment (swalke-ch) [inline src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:13]: Please update the  @JsonProperty for all the fields as per agreed json.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java [hunk 2]
```diff
@@ -10,8 +10,8 @@ public final class ACSP {
     @JsonProperty("payment_method")
     private String paymentMethod;
 
-    @JsonProperty("correspondence_address")
-    private Address correspondenceAddress;
+    @JsonProperty("service_address")
+    private ServiceAddress serviceAddress;
     @JsonProperty("office_address")
```
Comment (swalke-ch) [inline src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:15]: this needs to be "registered_office_address"

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java [hunk 3]
```diff
@@ -24,16 +24,12 @@ public final class ACSP {
...
     @JsonProperty("business_sector")
     private String businessSector;
-    @JsonProperty("amlmemberships")
-    private AmlMembership[] amlMemberships;
-    @JsonProperty("person_name")
-    private PersonName personName;
+    @JsonProperty("aml")
+    private Aml aml;
     @JsonProperty("business_name")
     private String businessName;
```
Comment (swalke-ch) [inline src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:30]: as we have "business_name" which corresponds to proposedCorporateBodyName tag in xml, I not sure what is the purpose of the field company_name ? sorry, if I am missing something

Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:30]: We get company_name in companyDetails object but business_name is basically used when we ask users if their business is registered with their name or some other name. Hence 2 different fields in front end.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java (hunk 2 above)
Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:15]: done

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java (hunk 3 above)
Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:30]: Yes, but does filling endpoint need this field? as it need to return data required for chips-filing-consumer. And we are not mapping company_name anywhere in json to xml conversion

**Review by ttingle-ch — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java (hunk 3 above)
Comment (sampankumar) [reply src/main/java/uk/gov/companieshouse/acsp/models/filing/ACSP.java:30]: Also we are populating company_name and business_name in different ways and I would like to keep it separate anyway just so if we have to make changes to one then other doesn't get affected.

**Review by swalke-ch — APPROVED** [review]

**Pull request merged**